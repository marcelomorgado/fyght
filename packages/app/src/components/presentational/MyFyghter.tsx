import React from "react";
import { Card } from "antd";
import FyghterRenamingModal from "./FyghterRenamingModal";
import FyghterChangeSkinModal from "./FyghterChangeSkinModal";
import { SkinAvatar } from "./SkinAvatar";

type Props = {
  fyghter: {
    id: number;
    skin: string;
    name: string;
    xp: number;
    qi: number;
    winCount: number;
    lossCount: number;
  };
};

// TODO: Edit fyghter name
export const MyFyghter = ({
  fyghter: { skin, name, xp, qi, winCount, lossCount },
}: Props) => {
  return (
    <Card
      type="inner"
      cover={<SkinAvatar skin={skin} size="medium" />}
      title={name}
    >
      <p></p>
      <p>{`wins: ${winCount} / losses: ${lossCount}`}</p>
      <p>{`XP: ${xp} / Qi: ${qi}`}</p>
      <FyghterRenamingModal />
      <p></p>
      <FyghterChangeSkinModal />
    </Card>
  );
};

export default MyFyghter;
