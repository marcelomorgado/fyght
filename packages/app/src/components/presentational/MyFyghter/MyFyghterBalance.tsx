import React from "react";
import { Spin } from "antd";
import { formatDai } from "../../../helpers";

interface Props {
  balance: BalanceState;
}

export const MyFyghterBalance: React.FC<Props> = ({ balance: { amount, loading } }: Props) => {
  return (
    <>
      <span style={{ color: "blue" }}>{`${formatDai(amount)}`}</span>{" "}
      {loading ? (
        <>
          {" "}
          <Spin size="small" />
        </>
      ) : null}
    </>
  );
};
