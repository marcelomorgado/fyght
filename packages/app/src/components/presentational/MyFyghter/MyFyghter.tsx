import React from "react";
import { Card } from "antd";
import { RenameFyghterButton } from "../RenameFyghterButton";
import { ChangeSkinButton } from "../ChangeSkinButton";
import { SkinAvatar } from "../SkinAvatar";
import { AvatarSize } from "../../../constants";
import { MyFyghterBalance } from "./MyFyghterBalance";
import { DepositButton } from "../DepositButton";
import { WithdrawAllButton } from "../WithdrawAllButton";

type Props = {
  fyghter: Fyghter;
};

export const MyFyghter: React.FC<Props> = ({ fyghter: { id: fyghterId, skin, name, xp, balance } }: Props) => {
  const isLoading = !fyghterId ? true : false;

  return (
    <Card type="inner" title={name} hoverable={true}>
      <SkinAvatar skin={skin} size={AvatarSize.MEDIUM} />
      <p></p>
      <p>{`XP: ${xp}`}</p>
      <p>
        {`Balance: `}
        <MyFyghterBalance value={balance} />
      </p>
      <RenameFyghterButton isLoading={isLoading} />
      <p></p>
      <ChangeSkinButton isLoading={isLoading} />
      <p></p>
      <DepositButton fyghterId={fyghterId} isLoading={isLoading} />
      <p></p>
      <WithdrawAllButton fyghterId={fyghterId} isLoading={isLoading} />
    </Card>
  );
};
