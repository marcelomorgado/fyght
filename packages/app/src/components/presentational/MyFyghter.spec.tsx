import * as React from "react";
import { shallow } from "enzyme";

import { MyFyghter } from "./MyFyghter";

describe("MyFyghter", () => {
  test("should render the component", () => {
    expect(
      shallow(
        <MyFyghter
          fyghter={{
            id: 1,
            name: "John",
            skin: "naked",
            xp: 2,
            winCount: 4,
            lossCount: 5,
          }}
        />
      )
    ).toMatchSnapshot();
  });
});
