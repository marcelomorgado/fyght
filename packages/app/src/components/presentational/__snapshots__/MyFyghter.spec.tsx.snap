// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyFyghter should render the component 1`] = `
<Card
  hoverable={true}
  title="John"
  type="inner"
>
  <Component
    size="medium"
    skin="naked"
  />
  <p />
  <p>
    wins: 4 / losses: 5
  </p>
  <p>
    XP: 2
  </p>
  <Component />
  <p />
  <Component />
</Card>
`;
